.test-infra/metrics/sync/github/sync_workflows.py

```diff
@@ -0,0 +1,162 @@
+Writing the latest 10 jobs of every postcommit workflow in master branch in a beammetrics database
+'''
+
+import os
+import sys
+import time
+import re
+import requests
+import psycopg2
+
+from datetime import datetime
+from github import GithubIntegration 
+
+DB_HOST = os.environ['DB_HOST']
+DB_PORT = os.environ['DB_PORT']
+DB_NAME = os.environ['DB_NAME']
+DB_USER_NAME = os.environ['DB_USER']
+DB_PASSWORD = os.environ['DB_PASS']
+GH_WORKFLOWS_TABLE_NAME = "github_workflows"
+# Number of workflows that fetch github API
+GH_NUMBER_OF_WORKFLOWS = 100  
+GH_WORKFLOWS_NUMBER_EXECUTIONS = 10
+GH_WORKFLOWS_YML_FILENAME = []
+
+# The table will save the latest ten run of every workflow
+GH_WORKFLOWS_CREATE_TABLE_QUERY = f"""
+CREATE TABLE IF NOT EXISTS {GH_WORKFLOWS_TABLE_NAME} (
+    job_name text PRIMARY KEY,
+    job_yml_filename text,
+    job1 text,
+    job2 text,
+    job3 text,
+    job4 text,
+    job5 text,
+    job6 text,
+    job7 text,
+    job8 text,
+    job9 text,
+    job10 text
+)
+"""
+def githubWorkflowsGrafanaSync(data,context):
+    print('Started')
+
+    print('Updating table with recent workflow runs')
+    databaseOperations(initDbConnection(),fetchWorkflowData())
+    print('Done')
+    return "Completed"
+
+def initDbConnection():
+    '''Init connection with the Database'''
+    connection = None
+    maxRetries = 3
+    i = 0 
+    while connection == None and i < maxRetries:
+        try:
+            connection = psycopg2.connect(
+                f"dbname='{DB_NAME}' user='{DB_USER_NAME}' host='{DB_HOST}'"
+                f" port='{DB_PORT}' password='{DB_PASSWORD}'")
+        except:
+            print('Failed to connect to DB; retrying in 1 minute')
+            sys.stdout.flush()
+            time.sleep(60)
+            i = i + 1
+    return connection
+
+def getToken():
+    git_integration = GithubIntegration(
+    os.environ["GH_APP_ID"],
+    os.environ["GH_PEM_KEY"])
+
+    token=git_integration.get_access_token(
+            os.environ["GH_APP_INSTALLATION_ID"]
+        ).token
+    return token
+
+def fetchWorkflowData():
+    '''Return a json with all the workflows and the latests
+    ten executions'''
+    listOfWorkflows = {}
+    workflowsStatus = {}
+    try:
+        url = "https://api.github.com/repos/apache/beam/actions/workflows"
+        queryOptions = { 'branch' : 'master', 'per_page' : GH_NUMBER_OF_WORKFLOWS }
+        response = requests.get(url = url, params = queryOptions)
+        jsonResponse = response.json()
+        workflows = jsonResponse['workflows']
+        for item in workflows:
+            path =(item['path'])
+            isPostCommit = re.search('(.*)postcommit(.*)',path)
+            if isPostCommit:
+                result = re.search('/(.*).yml', path)
+                path =(result.group(1)) + ".yml"
+                GH_WORKFLOWS_YML_FILENAME.append(path)
```

Comment:
Rather than having 2 objects, one for the workflow filenames and one for the list of workflows, why not just have an object that holds the name and the filenames?

Its not clear to me how we're ensuring that these will have the same order anyways